### DISCUSSION TITLE
Delete items/keys from list/structs

### DISCUSSION THREAD

**Discussion by vnzongzna** [cuelang/cue#519] [General]
Cue is perfect solution when we are thinking about the unification of items/structs. There are cases in our configuration when the end-user needs to delete items/keys from the existing list/struct like
```
DeleteThis: ["this", "that"]
useful_infra: {
	gcp: cluters:   10
	aws: clusters:  20
	this: clusters: 1
	that: clusters: 2
}
```
While one way to delete from `useful_infra` is
```
New_infra: {
    for k,v in useful_infra {
        if !list.Contains(DeleteThis, k) {
            "\(k)": v
        }
    }
}
```
Can we think of any other expression to do the disjunction as well?

**Comment by mpvl** [comment]
Would a "cast" operator as described in Issue #454 work for you?  Basically, instead of specifying what to remove, you would specify what to keep.

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/519.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
